src/GraphQLParser/ParserContext.Parse.cs

```diff
@@ -250,7 +249,7 @@ private GraphQLDirectiveDefinition ParseDirectiveDefinition()
             ExpectKeyword("on");
             var locations = ParseDirectiveLocations();
 
-            return _ignoreOptions == IgnoreOptions.IgnoreCommentsAndLocations
+            return _ignoreOptions.HasFlag(IgnoreOptions.Locations)
```

Comment:
Many of these conditions are wrong.  For instance, here if only 'Locations' is specified to be skipped, 'Comments' are also skipped because the non-"full" version is used which does not have a field to store the comments node.

```suggestion
            return _ignoreOptions == IgnoreOptions.All
```

Reply:
Tests should be added to ensure that each combination of ignored values is properly ignored or retained.

Reply:
I think actually the problem isn't that the tests don't exist, but that no `[InlineData(IgnoreOptions.Locations)]` were added to tests for comments

Reply:
Well, my mistake is obvious 🤦 .